main/acle.md

```diff
@@ -9546,25 +9680,2141 @@ possible to write these operations using normal C arithmetic. For example:
+  __attribute__((arm_streaming))
+  svcount_t svwhilehi_c8(uint64_t rn, uint64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilehs_c8(uint64_t rn, uint64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilele_c8(int64_t rn, int64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilelo_c8(uint64_t rn, uint64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilels_c8(uint64_t rn, uint64_t rm, uint64_t vl);
+  
+  
+  // Variants are also available for _c16, _c32 and _c64
+  __attribute__((arm_streaming))
+  svcount_t svwhilelt_c8(int64_t rn, int64_t rm, uint64_t vl);
+  
+  ```
+
+While (resulting in predicate tuple)
+
+``` c
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilege_b8_x2(int64_t rn, int64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilegt_b8_x2(int64_t rn, int64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilehi_b8_x2(uint64_t rn, uint64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilehs_b8_x2(uint64_t rn, uint64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilele_b8_x2(int64_t rn, int64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilelo_b8_x2(uint64_t rn, uint64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilels_b8_x2(uint64_t rn, uint64_t rm);
+  
+  
+  // Variants are also available for _b16, _b32 and _b64
+  __attribute__((arm_streaming_compatible))
+  svboolx2_t svwhilelt_b8_x2(int64_t rn, int64_t rm);
+  
+  ```
+
+#### SUNPK, UUNPK
+
+Multi-vector pack/unpack
+
+``` c
+  // Variants are also available for _u16_x2, _u32_x2, _s32_x2, _u64_x2 and _s64_x2
+  __attribute__((arm_streaming))
+  svint16x2_t svunpk[_s16_x2](svint8_t zn);
+  
+  
+  // Variants are also available for _u16_x4, _u32_x4, _s32_x4, _u64_x4 and _s64_x4
+  __attribute__((arm_streaming))
+  svint16x4_t svunpk[_s16_x4](svint8x2_t zn);
+  
+  ```
+
+#### ZIP
+
+Multi-vector zip.
+
+``` c
+  // Variants are also available for _u8, _u16, _s16, _f16, _bf16, _u32, _s32, _f32,
+  // _u64, _s64 and _f64
+  __attribute__((arm_streaming))
+  svint8x2_t svzip[_s8]_x2(svint8_t zn, svint8_t zm);
```

Comment:
This might be rehashing an old conversation, sorry, but why do these functions take individual vectors rather than x2 and x4 modes?  If the program does have individual vectors, it would be relatively simple to wrap them in `svcreate2` or `svcreate4` before passing them to the function.  But if the program already has a x2 or x4 mode, it would be a bit cumbersome to take them apart for the sake of the call.